File: 会计系统1/src/cn/edu/zust/controller/jizhangpingzheng.java
package cn.edu.zust.controller;

import java.io.IOException;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import net.sf.json.JSONObject;



public class jizhangpingzheng extends BaseServlet {

	private static final long serialVersionUID = 1L;

	public void save() throws IOException, SQLException{
		String jsonString = request.getParameter("jsonString");
		Map map = JSONObject.fromObject(jsonString);
		List<List> parameterList = getParameterList(map);
		
		//多条数据 插入多次
		for(List list : parameterList){
			String sql = "INSERT INTO accounting_voucher " +
					"VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
			runner.update(sql, list.toArray());
		}
		
	}
	//json
	private List getParameterList(Map map) {
		String pz_date = (String) map.get("pz_date");
		String pz_id = (String) map.get("pz_id");
		String kuaijizhuguan = (String) map.get("kuaijizhuguan");
		String jizhang = (String) map.get("jizhang");
		String jihe = (String) map.get("jihe");
		String zhidan = (String) map.get("zhidan");
		String chuna = (String) map.get("chuna");
		String danju = (String) map.get("danju");
		String jiaolingguanren = (String) map.get("jiaolingguanren");
		List data = (List) map.get("data");
		List parameterList = new ArrayList();
		for(int index=0;index<data.size();index++){
			List<String> list = new ArrayList<String>();
			list.add(pz_id);
			list.add(pz_date);
			list.add(kuaijizhuguan);
			list.add(jizhang);
			list.add(jihe);
			list.add(zhidan);
			list.add(chuna);
			list.add(jiaolingguanren);
			list.add(danju);
			Map m= (Map) data.get(index);
			list.add((String) m.get("zhaiyao"));
			list.add((String) m.get("jiesuanfangshi"));
			list.add((String) m.get("piaohao"));
			list.add((String) m.get("zongzhangkemu"));
			list.add((String) m.get("mingxikemu"));
			list.add((String) m.get("jiefangjine"));
			list.add((String) m.get("daifangjine"));
			parameterList.add(list);
			for(String s : list){
				System.out.print(s+" ");
			}
			System.out.println();
			/* list data
			 *  日期 凭证编号 会计主管人员 记账 稽核 制单 出纳 交领款人 单据张数 摘要1 结算 票号 总账科目 明细科目 11111110000 22222220000 
				日期 凭证编号 会计主管人员 记账 稽核 制单 出纳 交领款人 单据张数 摘要2 结算 票号 总账科目 明细科目 11111110000 22222220000 
				日期 凭证编号 会计主管人员 记账 稽核 制单 出纳 交领款人 单据张数 摘要3 结算 票号 总账科目 明细科目 11111110000 22222220000 
				日期 凭证编号 会计主管人员 记账 稽核 制单 出纳 交领款人 单据张数 摘要4 结算 票号 总账科目 明细科目 11111110000 22222220000 
			 * 
			 */
		}
		return parameterList;
	}

}

File: 会计系统1/WebRoot/js/jizhangpingzheng.js
$(document).ready(function(){
	//增加到data四行
	$(".table[name=model]").clone(true).insertBefore("#table2");
	$(".table[name=model]").clone(true).insertBefore("#table2");
	
//	$("#date").focus();
	$(".jine").children("input").focus(function(){
		$(this).val("");
	});
	$(".jine").keyup(function(e){
		if (e.keyCode >= 49 || e.keyCode<=50) {
			//按的很快时  bug修复
			if ($(this).children("input").val() != "") {
				$(this).next().children("input").focus();
			}
		}
	});
	$("#submit").click(function(){
		var pz_date = $("#pz_date").val();
		var pz_id = $("#pz_id").val();
		var kuaijizhuguan = $("#kuaijizhuguan").val();
		var jizhang = $("#jizhang").val();
		var jihe = $("#jihe").val();
		var zhidan = $("#zhidan").val();
		var chuna = $("#chuna").val();
		var jiaolingguanren = $("#jiaolingguanren").val();
		var danju = $(".danju>input").val();
		
		var zhaiyao = $(".zhaiyao>input");
		var jiesuanfangshi = $(".jiesuanfangshi>input");
		var piaohao = $(".piaohao>input");
		var zongzhangkemu = $(".zongzhangkemu>input");
		var mingxikemu = $(".mingxikemu>input");
		var jine = $(".jine>input");
		var arr = jine.map(function(){return this.value;});
		//借方金额贷方金额
		var arr2 = [];
		for(var i=0;i<jine.length/11;i++){
			var sum = 0;
			for(var j=i*11;j<i*11+11;j++){
				var t = arr[j]*1;
				sum = sum*10+t;
			}
			arr2.push(sum);
		}
		var data = [];
		for (var i = 0; i < zhaiyao.length; i++) {
			var t = {};
			t.zhaiyao = zhaiyao[i].value;
			t.jiesuanfangshi = jiesuanfangshi[i].value;
			t.piaohao = piaohao[i].value;
			t.zongzhangkemu = zongzhangkemu[i].value;
			t.mingxikemu = mingxikemu[i].value;
			t.jiefangjine = arr2[i*2]+"";
			t.daifangjine = arr2[i*2+1]+"";
			data.push(t);
		}
		var json = {};
		json.pz_date = pz_date;
		json.pz_id = pz_id;
		json.data = data;
		json.kuaijizhuguan = kuaijizhuguan;
		json.jizhang = jizhang;
		json.jihe = jihe;
		json.zhidan = zhidan;
		json.chuna = chuna;
		json.jiaolingguanren = jiaolingguanren;
		json.danju = danju;
		var jsonString = JSON.stringify(json);
		json.jsonString = jsonString;
		
		
		$.post("../jizhangpingzheng?method=save", json, function(data, textstatus){
			alert(data);
		});
		
		
	});
	
	
});
